pkg/resource/user/user_test.go

```diff
@@ -147,7 +150,78 @@ func TestUser_acceptance(t *testing.T) {
 			CheckNotExistsFunc:  checkNotExistsFunc,
 			CheckAttributesFunc: checkAttributesFunc,
 		},
+		{
+			Name:        "Create User with new password_sha256_hash field (OpenTofu version < 1.11 compatibility)",
+			ChEnv:       map[string]string{"CONFIGFILE": "config-single.xml"},
+			Protocol:    "native",
+			ClusterName: nil,
+			Resource: resourcebuilder.New(resourceType, resourceName).
+				WithStringAttribute("name", acctest.RandStringFromCharSet(10, acctest.CharSetAlphaNum)).
+				WithFunction("password_sha256_hash", "sha256", "changeme").
+				Build(),
+			ResourceName:        resourceName,
+			ResourceAddress:     fmt.Sprintf("%s.%s", resourceType, resourceName),
+			CheckNotExistsFunc:  checkNotExistsFunc,
+			CheckAttributesFunc: checkAttributesFunc,
+		},
+		{
+			Name:        "Create User with new password_sha256_hash field on cluster with localfile storage",
+			ChEnv:       map[string]string{"CONFIGFILE": "config-localfile.xml"},
+			Protocol:    "native",
+			ClusterName: &clusterName,
+			Resource: resourcebuilder.New(resourceType, resourceName).
+				WithStringAttribute("cluster_name", clusterName).
+				WithStringAttribute("name", acctest.RandStringFromCharSet(10, acctest.CharSetAlphaNum)).
+				WithFunction("password_sha256_hash", "sha256", "changeme").
+				Build(),
+			ResourceName:        resourceName,
+			ResourceAddress:     fmt.Sprintf("%s.%s", resourceType, resourceName),
+			CheckNotExistsFunc:  checkNotExistsFunc,
+			CheckAttributesFunc: checkAttributesFunc,
+		},
 	}
 
 	runner.RunTests(t, tests)
 }
+
+func TestUser_validation_acceptance(t *testing.T) {
+	providers := factories.ProviderFactories()
+	const sha256 = "e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855"
+
+	tests := []resource.TestCase{
+		{
+			ProtoV6ProviderFactories: providers,
+			Steps: []resource.TestStep{
+				// test ensure you can't use both password_sha256_hash and password_sha256_hash_wo
+				{
+					Config: fmt.Sprintf(`
+					resource "clickhousedbops_user" "test" {
+						name                    = "testuser"
+						password_sha256_hash    = "%s"
+						password_sha256_hash_wo = "%s"
+						password_sha256_hash_wo_version = 1
+					}
+				`, sha256, sha256),
+					PlanOnly:    true,
+					ExpectError: regexp.MustCompile(`Invalid Attribute Combination|one (and only one) of`),
+				},
+				// test ensure you can't use password_sha256_hash_wo without specifying password_sha256_hash_wo_version too
+				{
+					Config: fmt.Sprintf(`
+					resource "clickhousedbops_user" "test" {
+						name                    = "testuser"
+						password_sha256_hash    = "%s"
```

Comment:
Do you think this is needed? Either the description should be changed or this should be removed.